.github/workflows/proto_breaking.yml

```diff
@@ -0,0 +1,98 @@
+          chmod +x "${BAZEL}"
+          sudo mv "${BAZEL}" /usr/local/bin/bazel
+      - name: Mount bazel cache
+        uses: actions/cache@v4
+        with:
+          # See https://docs.bazel.build/versions/master/output_directories.html
+          path: "~/.cache/bazel"
+          # Create a new cache entry whenever Bazel files change.
+          # See https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows
+          key: bazel-${{ runner.os }}-build-${{ hashFiles('**/*.bzl', '**/*.bazel') }}
+          restore-keys: bazel-${{ runner.os }}-build-
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+      - name: Compare HEAD and BASE for breaking protobuf changes
+        run: |
+          python3 -m venv .venv
+          source .venv/bin/activate
+          pip install git+https://github.com/googleapis/proto-breaking-change-detector.git@v2.4.0
+          HEAD="${{ github.event.pull_request.head.sha }}"
+          BASE="$(git merge-base origin/main "${HEAD}")"
+          TMPDIR="$(mktemp -d -t compare-XXXX)"
+          TMPDIRBASE="${TMPDIR}/base"
+          TMPDIRHEAD="${TMPDIR}/head"
+
+          extract_protodesc() {
+            local TMPTARGETDIR
+            local TARGETS
+            local GENF
+            local TARGETDIR
+            local PROTONAME
+            TMPTARGETDIR="${1}"
+            TARGETS="$(bazel query 'attr("generator_function", "^proto_library$", "//...")' 2>/dev/null)"
+            GENF="$(bazel info bazel-genfiles)"
+            for TARGET in ${TARGETS}; do
+              TARGETDIR="$(echo "${TARGET}" | cut -d ":" -f1 | sed 's#^//##')"
+              PROTONAME="$(echo "${TARGET}" | cut -d ":" -f2-)"
+              mkdir -p "${TMPTARGETDIR}/${TARGETDIR}"
+              bazel build "${TARGET}"
+              cp "${GENF}/${TARGETDIR}/${PROTONAME}-descriptor-set.proto.bin" \
+                "${TMPTARGETDIR}/${TARGETDIR}/${PROTONAME}-descriptor-set.proto.bin"
+            done
+          }
+
+          bazel clean
+          extract_protodesc "${TMPDIRHEAD}"
+          git checkout "${BASE}"
+          bazel clean
+          BASETARGETS="$(bazel query 'attr("generator_function", "^proto_library$", "//...")' 2>/dev/null)"
+          extract_protodesc "${TMPDIRBASE}"
+
+          ERROR=false
+          for TARGET in ${BASETARGETS}; do
+            TARGETDIR="$(echo "${TARGET}" | cut -d ":" -f1 | sed 's#^//##')"
+            PROTONAME="$(echo "${TARGET}" | cut -d ":" -f2-)"
+            BASEPROTODESC="${TMPDIRBASE}/${TARGETDIR}/${PROTONAME}-descriptor-set.proto.bin"
+            HEADPROTODESC="${TMPDIRHEAD}/${TARGETDIR}/${PROTONAME}-descriptor-set.proto.bin"
+            if [ -f "${BASEPROTODESC}" ] && [ -f "${HEADPROTODESC}" ]; then
+              echo "Comparing build target ${TARGET} at HEAD and BASE..."
+              OUTPUT=$(proto-breaking-change-detector --original_descriptor_set_file_path="${BASEPROTODESC}" \
```

Comment:
Is there a way to make this a separate step from the set-up above to minimize noise when examining the output? e.g. in https://github.com/bstoll/gnoi/actions/runs/9486501478/job/26140974195?pr=2 the output is buried after a lot of logging from the set-up steps.

Reply:
I split up the steps - see new output example: https://github.com/bstoll/gnoi/actions/runs/9502889935/job/26192020815